Browser_Actions/Browser_ActionTests.cs

```diff
@@ -0,0 +1,126 @@
+            {
+                if (originalWindow != window)
+                {
+                    Driver.SwitchTo().Window(window);
+                    break;
+                }
+            }
+            wait.Until(wd => wd.Url == $"{MainUrl}/windows/new");
+            var elementContent = Driver.FindElement(By.ClassName("example"));
+            Assert.That(elementContent.Displayed);
+        }
+
+        [Test]
+        public void NavigateBackForward()
+        {
+            Setup();
+            var elementFormAuthentication = Driver.FindElement(By.CssSelector("[href*='login']"));
+            elementFormAuthentication.Click();
+            var elementUsername = Driver.FindElement(By.Id("username"));
+            elementUsername.SendKeys("tomsmith");
+            var elemenPassword = Driver.FindElement(By.Id("password"));
+            elemenPassword.SendKeys("SuperSecretPassword!");
+            var elementLoginButton = Driver.FindElement(By.XPath("//*[@class='radius']"));
+            elementLoginButton.Click();
+            var elementLogoutButton = Driver.FindElement(By.CssSelector("[href*='logout']"));
+            elementLogoutButton.Click();
+            Driver.Navigate().Back();
+            Driver.Navigate().Forward();
+            var pageUrl = Driver.Url;
+            Assert.That(pageUrl, Is.EqualTo($"{MainUrl}/login"));
+            var elementContentLogin = Driver.FindElement(By.Id("content"));
+            Assert.That(elementContentLogin.Text.Contains("Login Page"));
+        }
+
+        [Test]
+        public void NavigateToUrlAndRefresh()
+        {
+            Setup();
+            var elementDynamicLoading = Driver.FindElement(By.CssSelector("[href*='dynamic_loading']"));
+            elementDynamicLoading.Click();
+            var elementExample = Driver.FindElement(By.XPath("//*[@id='content']//a[1]"));
+            elementExample.Click();
+            var elementStartButton = Driver.FindElement(By.TagName("button"));
+            elementStartButton.Click();
+            var wait = new WebDriverWait(Driver, TimeSpan.FromSeconds(20));
+            Assert.That(wait.Until(ExpectedConditions.ElementIsVisible(By.Id("finish"))).Displayed);
+            var currentUrl = Driver.Url;
+            Setup();
+            var homePageUrl = Driver.Url;
+
+            Assert.False(homePageUrl.Equals(currentUrl));
+            Assert.That(homePageUrl, Is.EqualTo($"{MainUrl}/"));
+            Driver.Navigate().Refresh();
+        }
+
+        [Test]
+        public void MaximizeWindowChangeWindowSize()
+        {
+            Setup();
+            var elementLargeDeepDom = Driver.FindElement(By.CssSelector("[href*='large']"));
+            elementLargeDeepDom.Click();
+            var elementLastDom = Driver.FindElement(By.XPath("//*[@id='large-table']/tbody/tr[50]/td[50]"));
+            ((IJavaScriptExecutor)Driver).ExecuteScript("arguments[0].scrollIntoView();", elementLastDom);
```

Comment:
You use js, you can try to use Actions - it is the same, but from Selenium

Reply:
left js, since Actions give error with Firefox browser